### DISCUSSION TITLE
node:http ( make request and response ) browser compatible

### DISCUSSION THREAD

**Discussion by PodaruDragos** [nodejs/help#4729] [Ideas]
Hello guys ,

Firstly thanks for adding more "browser-like" apis to `node` itself 👍 ( also features like native watch and test. I really like where the `node` ecosystem is heading 👍 ) . 
This is mostly a question rather than an idea.

I understand this is clearly breaking, but aligning the `req: Request, res: Response` with the browser versions will allow us to unify the experience including in (`deno`, `bun`) and will potentially unlock great features natively like 

`response.json()`
`request.formData()`

I did not find any indications online if this is a thing worth considering so here goes: 
Are there any plans to consider adding this ?

**Comment by ThaUnknown** [comment]
I like how [CloudFlare handled this](https://developers.cloudflare.com/workers/learning/service-worker/), with http servers using [W3C's service worker syntax](https://www.w3.org/TR/service-workers/#serviceworkerglobalscope-interface), maybe the http[s] module[s] could also implement a service worker like syntax and handling for this?
